libs/hmpb/src/ballot_templates/nh_ballot_template.tsx

```diff
@@ -218,6 +243,374 @@ function BallotPageFrame({
+    const contestElements = section.map((contest) => (
+      <Contest key={contest.id} contest={contest} election={election} />
+    ));
+    const numColumns = section[0].type === 'candidate' ? 3 : 2;
+    const columnWidthPx =
+      (dimensions.width - horizontalGapPx * (numColumns - 1)) / numColumns;
+    const contestMeasurements = await scratchpad.measureElements(
+      <BackendLanguageContextProvider
+        currentLanguageCode={primaryLanguageCode(ballotStyle)}
+        uiStringsPackage={election.ballotStrings}
+      >
+        {contestElements.map((contest, i) => (
+          <div
+            className="contestWrapper"
+            key={i}
+            style={{ width: `${columnWidthPx}px` }}
+          >
+            {contest}
+          </div>
+        ))}
+      </BackendLanguageContextProvider>,
+      '.contestWrapper'
+    );
+    const measuredContests = iter(contestElements)
+      .zip(contestMeasurements)
+      .map(([element, measurements]) => ({ element, ...measurements }))
+      .toArray();
+
+    const { columns, height, leftoverElements } = layOutInColumns({
+      elements: measuredContests,
+      numColumns,
+      maxColumnHeight: dimensions.height - heightUsed,
+      elementGap: verticalGapPx,
+    });
+
+    // Put leftover elements back on the front of the queue
+    if (leftoverElements.length > 0) {
+      contestSectionsLeftToLayout.unshift(
+        leftoverElements.map(({ element }) => element.props.contest)
+      );
+    }
+
+    // If there wasn't enough room left for any contests, go to the next page
+    if (height === 0) {
+      break;
+    }
+
+    // Add vertical gap to account for space between sections
+    heightUsed += height + verticalGapPx;
+    pageSections.push(
+      <div
+        key={`section-${pageSections.length + 1}`}
+        style={{ display: 'flex', gap: `${horizontalGapPx}px` }}
+      >
+        {columns.map((column, i) => (
+          <div
+            key={`column-${i}`}
+            style={{
+              flex: 1,
+              display: 'flex',
+              flexDirection: 'column',
+              gap: `${verticalGapPx}px`,
+            }}
+          >
+            {column.map(({ element }) => element)}
+          </div>
+        ))}
+      </div>
+    );
+  }
+
+  const currentPageElement =
+    pageSections.length > 0 ? (
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          gap: `${verticalGapPx}px`,
+        }}
+      >
+        {pageSections}
+      </div>
+    ) : (
+      <BlankPageMessage />
+    );
+  const nextPageProps =
+    contestSectionsLeftToLayout.length > 0
+      ? {
+          ...restProps,
+          ballotStyleId,
+          election: {
+            ...election,
+            contests: contestSectionsLeftToLayout.flat(),
+          },
+        }
+      : undefined;
+
+  return {
+    currentPageElement,
+    nextPageProps,
+  };
+}
```

Comment:
Very possibly already on your radar but given that the NH template will always have right-aligned bubbles now, I think we can probably remove the placeholder toggle and feature flag we added for left/right bubble alignment in https://github.com/votingworks/vxsuite/pull/5811

Reply:
Thanks for the reminder!